amgut/lib/data_access/survey.py

```diff
@@ -194,18 +194,22 @@ class Group(object):
     """
     _group_table = 'survey_group'
     _group_questions_table = 'group_questions'
+    _questions_table = 'survey_question'
 
     def __init__(self, ID):
         self.id = ID
         n = self.american_name
         qs = [Question.factory(x[0], n) for x in db_conn.execute_fetchall('''
-            select gq.survey_question_id
-            from {0} sg join {1} gq on sg.group_order = gq.survey_group
-            where sg.group_order = %s
-            order by gq.display_index
+            SELECT gq.survey_question_id
+            FROM {0} sg
+            JOIN {1} gq ON sg.group_order = gq.survey_group
+            LEFT JOIN {2} sq USING (survey_question_id)
+            WHERE sg.group_order = %s AND sq.retired = FALSE
```

Comment:
I like this a lot as it naturally supports future retired questions if they arise.